app/src/main/java/com/infomaniak/mail/ui/main/folder/ThreadListViewModel.kt

```diff
@@ -66,18 +66,34 @@ class ThreadListViewModel @Inject constructor(
     }
 
     fun checkWebViewVersion(canShowWebViewOutdated: Boolean) {
+        val webViewPackage =  WebViewCompat.getCurrentWebViewPackage(appContext)
+        val webViewPackageName =  webViewPackage?.packageName
         val (webViewVersionName, webViewMajorVersion) =
-            WebViewCompat.getCurrentWebViewPackage(appContext)?.versionName?.let { versionName ->
+            webViewPackage?.versionName?.let { versionName ->
                 val majorVersion = runCatching { versionName.substringBefore('.').toInt() }.getOrDefault(0)
                 versionName to majorVersion
             } ?: (null to 0)
 
-        val badMajorVersion = webViewMajorVersion < WEBVIEW_MIN_VERSION
-        if (badMajorVersion) SentryDebug.sendWebViewVersionName(webViewVersionName, webViewMajorVersion)
+        fun hasBadMajorVersion(): Boolean {
+            return when (webViewPackageName) {
+                WEBVIEW_OFFICIAL_PACKAGE_NAME -> {
+                    webViewMajorVersion < WEBVIEW_OFFICIAL_MIN_VERSION
+                }
+                else -> {
+                    // We'll add other package names in the future if needed here
+                    false
+                }
+            }
+        }
```

Comment:
I think we usually put the inner functions on top of the impacted scope. I don't find it really readable like this.

Reply:
I'd rewrite the entire function like this : 
```suggestion
        fun checkWebViewVersion(canShowWebViewOutdated: Boolean) {
        val webViewPackage = WebViewCompat.getCurrentWebViewPackage(appContext)
        val webViewPackageName = webViewPackage?.packageName

        val (webViewVersionName, webViewMajorVersion) = webViewPackage?.versionName?.let { versionName ->
            val majorVersion = runCatching { versionName.substringBefore('.').toInt() }.getOrDefault(defaultValue = 0)
            versionName to majorVersion
        } ?: (null to 0)

        fun hasOutdatedMajorVersion(): Boolean = when (webViewPackageName) {
                WEBVIEW_OFFICIAL_PACKAGE_NAME -> webViewMajorVersion < WEBVIEW_OFFICIAL_MIN_VERSION
                else -> false // We'll add other package names in the future if needed here
        }

        // TODO: (23/07) Remove this log in a few weeks/month if we don't have any Sentry anymore
        val hasOudatedMajorVersion = hasOutdatedMajorVersion()
        if (hasOudatedMajorVersion) SentryDebug.sendWebViewVersionName(webViewPackageName, webViewVersionName, webViewMajorVersion)
        isWebViewOutdated.value = canShowWebViewOutdated && hasOudatedMajorVersion
    }
```

Reply:
I kept the inner function here, but I'm not sure about its utility, I think we could just `val hasOutdatedVersion = when {...`